alacritty/src/display/window.rs

```diff
@@ -451,16 +449,12 @@ impl Window {
     /// This prevents rendering artifacts from showing up when the window is transparent.
     #[cfg(target_os = "macos")]
     pub fn set_has_shadow(&self, has_shadows: bool) {
-        let ns_view = match self.raw_window_handle() {
-            RawWindowHandle::AppKit(handle) => handle.ns_view.as_ptr() as id,
+        let view = match self.raw_window_handle() {
+            RawWindowHandle::AppKit(handle) => unsafe { handle.ns_view.cast::<NSView>().as_ref() },
```

Comment:
Is this guaranteed to be safe? If so, is there a reason this isn't made public through a safe API?

Reply:
> Safe as long as you're on the main thread.

And where does this function guarantee that it's being called from the main thread?

Reply:
Right, yeah, it doesn't. I think you have an assertion for this already (`raw_window_handle` will panic if it's not on the main thread), but we should do this explicitly too.